akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransport.scala

```diff
@@ -57,11 +56,15 @@ object ClientTransport {
    */
   def httpsProxy(proxyAddress: InetSocketAddress): ClientTransport = new HttpsProxyTransport(proxyAddress)
 
-  private case class HttpsProxyTransport(proxyAddress: InetSocketAddress, underlyingTransport: ClientTransport = TCP) extends ClientTransport {
+  def httpsProxy(proxyAddress: InetSocketAddress, proxyAuth: HttpCredentials): ClientTransport = new HttpsProxyTransport(proxyAddress, proxyAuth = Some(proxyAuth))
```

Comment:
Yep, copying would be fine adding a note about the credentials.



> Also, the ClientTransport object already has the @apimaychange marker, so do I also need to add it on the method?

I see, then it's ok without an extra marker for the method.